[CMake] Build with `CIRCT_SLANG_FRONTEND_ENABLED` fails to configure

Switching on the `CIRCT_SLANG_FRONTEND_ENABLED` option prevents my local builds from configuring:

```
-- Building slang from source
-- slang version: 3.0.firtool-1.64.0-21-g15594935e+firtool-1.64.0-21-g15594935e
CMake Error at build/_deps/slang-src/CMakeLists.txt:51 (project):
  VERSION "3.0.firtool-1.64.0-21-g15594935e" format invalid.
```
Since the CIRCT CMake script fetches slang's release tarball instead of the git repository, slang's [gitversion.cmake](https://github.com/MikePopoloski/slang/blob/v3.0/cmake/gitversion.cmake) script runs into the CIRCT repository, performs `git describe` and tries to parse this as its version.
Checking the CI logs, the CIRCT commit hash appears in the slang version string. My best guess here is that the CI does not fetch the tags, so this issue is not triggered.

My best workaround so far is to have CMake fetch the repository instead of the tarball, as it was originally in #6620. But maybe somene has a better idea.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -546,9 +546,9 @@ if(CIRCT_SLANG_FRONTEND_ENABLED)
     include(FetchContent)
     FetchContent_Declare(
       slang
-      URL https://github.com/MikePopoloski/slang/archive/refs/tags/v3.0.tar.gz
-      URL_HASH SHA256=02d184133525f6330a32bef492761b8ff52d0a33caed08eb51733f63cddb08e3
-    )
+      GIT_REPOSITORY https://github.com/MikePopoloski/slang.git
+      GIT_TAG v3.0
+      GIT_SHALLOW ON)
     set(FETCHCONTENT_TRY_FIND_PACKAGE_MODE "NEVER")
 
     # Force Slang to be built as a static library to avoid messing around with

